vlm/src/main/scala/geotrellis/contrib/vlm/RasterSource.scala

```diff
@@ -102,7 +102,12 @@ trait RasterSource extends CellGrid[Long] with Serializable {
     * @group reproject a
     */
   def reprojectToGrid(crs: CRS, grid: GridExtent[Long], method: ResampleMethod = NearestNeighbor, strategy: OverviewStrategy = AutoHigherResolution): RasterSource =
-    reproject(crs, Reproject.Options(method = method, parentGridExtent = Some(grid)), strategy)
+    if (crs == this.crs && grid == this.gridExtent)
+      this
+    else if (crs == this.crs)
+      resampleToGrid(grid)
```

Comment:
I'm unclear on whether doing the 'smart' thing here is actually desirable. It might make the API more magical

Reply:
i'm still looking into this PR; however, answering your comment; I think in general idea - it totally fine aligns with this issue https://github.com/geotrellis/geotrellis-contrib/issues/102